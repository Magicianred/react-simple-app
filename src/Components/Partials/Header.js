import React from 'react';
import PropTypes from 'prop-types';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import logo from './../../logo.svg';

/**
 * Component for showing the header of the site.
 * @component
 * @example
 * const sitename = 'my site name'
 * return (
 *   <Header title={sitename} />
 * )
 */
const Header = ({ title }) => {
    return (
        <header className="partial header">
            <>
                <Navbar bg="light">
                    <Navbar.Brand href="/">
                    <img
                        src={logo}
                        width="30"
                        height="30"
                        className="d-inline-block align-top"
                        alt={`${title} logo`}
                    />
                    </Navbar.Brand>
                    <Navbar.Toggle aria-controls="basic-navbar-nav" />
                    <Navbar.Collapse id="basic-navbar-nav">
                        <Nav className="mr-auto">
                            <Nav.Link href="/">Home</Nav.Link>
                            <Nav.Link href="/whoare">Who Are</Nav.Link>
                            <Nav.Link href="/contact">Contact Us</Nav.Link>
                            <Nav.Link href="/altro">Altro</Nav.Link>
                            <Nav.Link target="_blank" href="/docs">Docs [open new page]</Nav.Link>
                        </Nav>
                    </Navbar.Collapse>
                </Navbar>
            </>
        </header>
    );
}

Header.propTypes = {
    /**
     * Title of the site
     */
    title: PropTypes.string.isRequired,
}

Header.defaultProps = {
    title: 'Default Site title'
}

export default Header;