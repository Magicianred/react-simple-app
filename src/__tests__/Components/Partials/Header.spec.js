import React from 'react';
import { create } from 'react-test-renderer';

import Header from './../../../Components/Partials/Header'

describe("Header Partials Component", () => {

    describe("Snapshots", () => {
    
        it("Matches the snapshot (with default title)", () => {
            const instance = create(<Header />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
        it("Matches the snapshot (with title: 'My App Title')", () => {
            const instance = create(<Header title="My App Title" />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    
});