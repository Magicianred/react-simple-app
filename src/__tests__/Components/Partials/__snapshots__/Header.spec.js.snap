// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Partials Component Snapshots Matches the snapshot (with default title) 1`] = `
<header
  className="partial header"
>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Default Site title logo"
        className="d-inline-block align-top"
        height="30"
        src="logo.svg"
        width="30"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/whoare"
          href="/whoare"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Who Are
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/contact"
          href="/contact"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Contact Us
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/altro"
          href="/altro"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Altro
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/docs"
          href="/docs"
          onClick={[Function]}
          onKeyDown={[Function]}
          target="_blank"
        >
          Docs [open new page]
        </a>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header Partials Component Snapshots Matches the snapshot (with title: 'My App Title') 1`] = `
<header
  className="partial header"
>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="My App Title logo"
        className="d-inline-block align-top"
        height="30"
        src="logo.svg"
        width="30"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/whoare"
          href="/whoare"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Who Are
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/contact"
          href="/contact"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Contact Us
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/altro"
          href="/altro"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Altro
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/docs"
          href="/docs"
          onClick={[Function]}
          onKeyDown={[Function]}
          target="_blank"
        >
          Docs [open new page]
        </a>
      </div>
    </div>
  </nav>
</header>
`;
